OSX CI build failed but shown as 'passed'

CI build Is failed with the next error:

```
Linking boinc
ld: symbol(s) not found for architecture x86_64

clang: error: linker command failed with exit code 1 (use -v to see invocation)

** BUILD FAILED **

The following build commands failed:

	Ld /Users/runner/work/boinc/boinc/mac_build/build/boinc.build/Deployment/BOINC_Client.build/Objects-normal/x86_64/Binary/boinc normal x86_64

(1 failure)
```

https://github.com/BOINC/boinc/runs/4912711265?check_suite_focus=true

But build script ignores this.

@CharlieFenton, could you please take a look and fix both script and build?

Thanks in advance.

--- a/mac_build/BuildMacBOINC.sh
+++ b/mac_build/BuildMacBOINC.sh
@@ -2,7 +2,7 @@
 
 # This file is part of BOINC.
 # http://boinc.berkeley.edu
-# Copyright (C) 2017 University of California
+# Copyright (C) 2022 University of California
 #
 # BOINC is free software; you can redistribute it and/or modify it
 # under the terms of the GNU Lesser General Public License
@@ -35,6 +35,7 @@
 # Updated 10/14/18 for Xcode 10 (use this script only with BOINC 7.15 or later)
 # Updated 3/31/21 To eliminate redundant -c++11 arg since C++11 build is now standard
 # Updated 5/19/21 for compatibility with zsh
+# Updated 7/12/22 result is moved out of eval string to get correct status on CI if build fails
 #
 ## This script requires OS 10.8 or later
 #
@@ -171,8 +172,9 @@ for f in "${settings[@]}"; do
 done
 
 ## For unknown reasons, this xcodebuild call generates syntax errors under zsh
-## unless we enclose the command in quotes and invoke it with eval. 
-eval "xcodebuild -project boinc.xcodeproj ${targets} -configuration ${style} -sdk \"${SDKPATH}\" ${doclean} build ${uselibcplusplus} ${theSettings};result=$?"
+## unless we enclose the command in quotes and invoke it with eval.
+eval "xcodebuild -project boinc.xcodeproj ${targets} -configuration ${style} -sdk \"${SDKPATH}\" ${doclean} build ${uselibcplusplus} ${theSettings}"
+result=$?
 
 if [ $result -eq 0 ]; then
     # build ibboinc_zip.a for -all or -lib or default, where

